blocks/library/freeform/index.js

```diff
@@ -13,22 +56,88 @@ registerBlock( 'core/freeform', {
 	category: 'common',
 
 	attributes: {
-		html: html(),
+		content: children(),
+		listType: internalValue( null ),
+		inBlockQuote: internalValue( false ),
+		editor: internalValue( null ),
+	},
+
+	defaultAttributes: {
+		content: <p />,
+		listType: null,
+		inBlockQuote: false,
+		editor: null,
 	},
 
-	edit( { attributes } ) {
+	controls: [
+		{
+			icon: 'editor-quote',
+			title: wp.i18n.__( 'Quote' ),
+			isActive: blockQuoteIsActive(),
+			onClick: execCommand( 'mceBlockQuote' ),
+		},
+		{
+			icon: 'editor-ul',
+			title: wp.i18n.__( 'Convert to unordered' ),
+			isActive: listIsActive( 'UL' ),
+			onClick: execCommand( 'InsertUnorderedList' ),
+		},
+		{
+			icon: 'editor-ol',
+			title: wp.i18n.__( 'Convert to ordered' ),
+			isActive: listIsActive( 'OL' ),
+			onClick: execCommand( 'InsertOrderedList' ),
+		},
+	],
+
+	advControls: [
+		{
+			icon: 'editor-outdent',
+			title: wp.i18n.__( 'Outdent list item' ),
+			onClick: execCommand( 'Outdent' ),
+		},
+		{
+			icon: 'editor-indent',
+			title: wp.i18n.__( 'Indent list item' ),
+			onClick: execCommand( 'Indent' ),
+		},
+	],
+
+	edit( { attributes, setAttributes, focus, setFocus } ) {
+		const { content } = attributes;
+
 		return (
-			<div
-				contentEditable
-				suppressContentEditableWarning
-			>
-				{ attributes.html }
-			</div>
+			<Editable
```

Comment:
Should we use `Editable` for this block? I don't know the answer here because we may think this block would be way more complex than the regular `Editable` usage. In the same time having access to the `onSetup` callback allows us to enhance the behaviour, so maybe we could try for now and reconsider later.